src/lem/circuit.rs

```diff
@@ -584,9 +657,6 @@ fn synthesize_block<F: LurkField, CS: ConstraintSystem<F>, C: Coprocessor<F>>(
                 let not_dummy_and_not_blank = not_dummy.get_value() == Some(true) && !ctx.blank;
                 let collected_z_ptrs = if not_dummy_and_not_blank {
                     let collected_ptrs = ctx.bindings.get_many_ptr(out)?;
-                    if out.len() != collected_ptrs.len() {
-                        bail!("Incompatible output length for coprocessor {sym}")
-                    }
```

Comment:
We can at least leave an `assert_eq!(out.len(), collected_ptrs.len());`

Reply:
There's no need to check this, because `collected_ptrs` is given by `get_many_ptr` on `out`, which is guaranteed to return the same number of variables